### PR TITLE
Fix Duplicate Gym Pokemon

### PR DESCRIPTION
**Pull request by maddhatter**
## Description

Fixes duplicate Pokemon issue from #666 (for real)
## How Has This Been Tested?

I bastardized my setup to force duplicate Pokemon to get inserted, then tested these changes to confirm duplicates no longer exist.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by gabrielsoldani** [comment]
Any reason why ('gym_id', 'pokemon_uid') is a composite unique index instead of the more obvious composite primary key? i.e.

```python
class Meta:
        primary_key = CompositeKey('gym_id', 'pokemon_uid')
```

**Comment by maddhatter** [comment]
Originally, it was so that the table wouldn't need to be recreated (Peewee migrations don't support changing the PK), but I changed it to recreate the table anyway... So, it's a composite key now.

DeleteQuery was gone because I apparently had a mild stroke while writing this or something... it's back now.

File: pogom/models.py [hunk 1]
```diff
@@ -940,3 +937,6 @@ def database_migrate(db, old_ver):
             migrator.drop_column('gymdetails', 'description'),
             migrator.add_column('gymdetails', 'description', TextField(null=True, default=""))
         )
+
+    if old_ver < 8 and old_ver >= 6:  # only do this if the GymMember table exists, it was introduced in v6
+        db.drop_tables([GymMember])
```
Comment (chuyskywalker) [inline pogom/models.py]: What re-creates the table?

Comment (maddhatter) [inline pogom/models.py]: The migrations get called from the [`create_tables`](https://github.com/PokemonGoMap/PokemonGo-Map/blob/develop/pogom/models.py#L855) method right before the actual table creation gets run immediately after the migrations (`database_migrate` gets called from `verify_database_schema`).

**Comment by FrostTheFox** [comment]
Got assigned this, gonna try to get some feedback/testing on it and see how its doing.

**Comment by jtp10181** [comment]
I installed this to try and was getting no gym member info anymore. The gymmember table is not even populating. I should have backed up my DB before this, hopefully I can get it back to working again.

**Comment by sebastienvercammen** [comment]
We're cleaning up some old PRs that have been left around from before we had the time to handle everything properly. With the last few weeks going fantastically well, cleaning up the PR and issue queue will let us move forward again :) If this PR is still valid, don't hesitate to re-open and poke us.

**Pull request closed without merging**